hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -43,16 +43,15 @@ String::String(const __FlashStringHelper *pstr)
 	*this = pstr;
 }
 
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if STRING_HAS_MOVE
 String::String(String &&rval)
+	: buffer(rval.buffer)
+	, capacity(rval.capacity)
+	, len(rval.len)
```

Comment:
Even if std::move was available it would not make much sense to use it here - all it does is to cast is argument to rvalue reference. For integral types and pointers move construction is same as copy construction.

I agree that it would be useful to have move and forward available in lib.